src/ZohoDatabaseSyncZoho.php

```diff
@@ -0,0 +1,225 @@
+<?php
+
+namespace Wabel\Zoho\CRM\Copy;
+
+use Doctrine\DBAL\Connection;
+use Psr\Log\LoggerInterface;
+use Psr\Log\NullLogger;
+use Wabel\Zoho\CRM\AbstractZohoDao;
+use Wabel\Zoho\CRM\ZohoBeanInterface;
+use Wabel\Zoho\CRM\Exception\ZohoCRMException;
+
+/**
+ * Description of ZohoDatabaseSyncZoho
+ *
+ * @author rbergina
+ */
+class ZohoDatabaseSyncZoho
+{
+
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     *
+     * @var string
+     */
+    private $prefix;
+
+    /**
+     * @param Connection $connection
+     * @param string $prefix
+     * @param LoggerInterface $logger
+     */
+    public function __construct(Connection $connection, $prefix = 'zoho_', LoggerInterface $logger = null)
+    {
+        $this->connection = $connection;
+        $this->prefix = $prefix;
+        if ($logger === null) {
+            $this->logger = new NullLogger();
+        } else {
+            $this->logger = $logger;
+        }
+    }
+
+    /**
+     *
+     * @param AbstractZohoDao $zohoDao
+     * @return array
+     */
+    private function findMethodValues(AbstractZohoDao $zohoDao){
+        $fieldsMatching = array();
+        foreach ($zohoDao->getFields() as $fieldsDescriptor) {
+            foreach (array_values($fieldsDescriptor) as $fieldDescriptor) {
+                $fieldsMatching[$fieldDescriptor['name']] = [
+                    'setter' => $fieldDescriptor['setter'],
+                    'type' => $fieldDescriptor['type']
+                ];
+            }
+
+        }
+        return $fieldsMatching;
+    }
+
+    /**
+     * Insert or Update rows.
+     * @param AbstractZohoDao $zohoDao
+     * @param string $localTable
+     */
+    public function pushDataToZoho(AbstractZohoDao $zohoDao, $localTable, $update = false){
```

Comment:
This method should be private.
Also, the signature is weird. $localTable can be guessed from the $update parameter.
We should really drop this parameter and replace it with:

$localTable = $update ? 'local_update' : 'local_insert';